### PR TITLE
[charts] Test pointer events

### PR DESCRIPTION
**Pull request by alexfauquette**
Part of #13611

For a reason I ignore, the mouse position has a weird conversion. When I set event.clientX=201 I get a SVG position of 196

### CODE REVIEW

**Comment by alexfauquette** [comment]
@JCQuintas Could have a first look. Once we aggree on the testing for bar, I will add other charts with the same stategy

File: packages/x-charts/src/BarChart/checkClickEvent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    it('should provide the accurate context as second argument', function test() {
+      if (isJSDOM) {
+        // can't do Pointer event with JSDom https://github.com/jsdom/jsdom/issues/2527
+        this.skip();
+      }
+      const onAxisClick = spy();
+      render(
+        <div
+          style={{
+            margin: -8, // No idea why, but that make the SVG coordinates match the HTML coordinates
```
Comment (JCQuintas) [inline packages/x-charts/src/BarChart/checkClickEvent.test.tsx:80]: This happens because body has a default margin of `8px`

**Comment by JCQuintas** [comment]
@alexfauquette sounds good

**Review by JCQuintas — APPROVED** [review]

**Comment by JCQuintas** [comment]
Seems like the test is not cleaning the dom after it runs?

**Comment by alexfauquette** [comment]
> Seems like the test is not cleaning the dom after it runs?

Effectively, to measure the text size, we add a span in an imperative way.

File: packages/x-charts/src/internals/domUtils.ts [hunk 2]
```diff
@@ -152,3 +152,9 @@ export const getStringSize = (text: string | number, style: React.CSSProperties
     return { width: 0, height: 0 };
   }
 };
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function unstable_cleanupDOM() {
+  const measurementSpan = document.getElementById(MEASUREMENT_SPAN_ID);
+  measurementSpan?.remove();
+}
```
Comment (JCQuintas) [inline packages/x-charts/src/internals/domUtils.ts:160]: Cant we empty the innertext or remove the element before returning the results of `getStringSize` instead?

Comment (JCQuintas) [reply packages/x-charts/src/internals/domUtils.ts]: That seems to have broken the pipeline 😅

Comment (alexfauquette) [reply packages/x-charts/src/internals/domUtils.ts]: No, that was just some outrage on GitHub ;)

**Pull request merged**